base/applications/mspaint/dialogs.cpp

```diff
@@ -135,20 +135,22 @@ LRESULT CAttributesDialog::OnDefault(WORD wNotifyCode, WORD wID, HWND hWndCtl, B
 
 LRESULT CAttributesDialog::OnRadioButton1(WORD wNotifyCode, WORD wID, HWND hWndCtl, BOOL& bHandled)
 {
+    FixDpi();
     CString strNum;
-    strNum.Format(_T("%.3lf"), newWidth / (0.0254 * fileHPPM));
+    strNum.Format(_T("%.3lf"), newWidth / g_xDpi);
     SetDlgItemText(IDD_ATTRIBUTESEDIT1, strNum);
-    strNum.Format(_T("%.3lf"), newHeight / (0.0254 * fileVPPM));
+    strNum.Format(_T("%.3lf"), newHeight / g_yDpi);
     SetDlgItemText(IDD_ATTRIBUTESEDIT2, strNum);
     return 0;
 }
 
 LRESULT CAttributesDialog::OnRadioButton2(WORD wNotifyCode, WORD wID, HWND hWndCtl, BOOL& bHandled)
 {
+    FixDpi();
     CString strNum;
-    strNum.Format(_T("%.3lf"), newWidth * 100.0 / fileHPPM);
+    strNum.Format(_T("%.3lf"), newWidth / (PpmFromDpi(g_xDpi) / 100));
     SetDlgItemText(IDD_ATTRIBUTESEDIT1, strNum);
-    strNum.Format(_T("%.3lf"), newHeight * 100.0 / fileVPPM);
+    strNum.Format(_T("%.3lf"), newHeight / (PpmFromDpi(g_yDpi) / 100));
```

Comment:
Won't this way of doing the division create rounding errors?

Reply:
The return value of `PpmFromDpi` is a `float`.

Reply:
Improved.